Towny/src/main/java/com/palmergames/bukkit/towny/tasks/TownClaim.java

```diff
@@ -78,7 +80,7 @@ public void run() {
 				TownyMessaging.sendMsg(player, claim ? Translatable.of("msg_process_town_claim") : Translatable.of("msg_process_town_unclaim"));
 
 			if (selection != null) // Selection is never null unless a resident has done /t unclaim all.
-				processSelection();
+				processSelection(isOverClaim);

```

Comment:
instead of passing the variable on through the methods just use `isOverClaim` variable when firing the events.